src/XRP/model/xrp-deposit-preauth.ts

```diff
@@ -21,12 +22,19 @@ export default class XrpDepositPreauth {
   public static from(
     depositPreauth: DepositPreauth,
     xrplNetwork: XrplNetwork,
-  ): XrpDepositPreauth | undefined {
+  ): XrpDepositPreauth {
     const authorize = depositPreauth.getAuthorize()?.getValue()?.getAddress()
     const unauthorize = depositPreauth
       .getUnauthorize()
       ?.getValue()
       ?.getAddress()
+    // You must provide either Authorize or Unauthorize, but not both.
```

Comment:
i think you can remove this - the logic + console message express this invariant clearly imo